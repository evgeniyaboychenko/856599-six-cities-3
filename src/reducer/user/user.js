import {extend} from '../../utils/utils.js';
import {adaptUser} from '../../utils/adapter.js';
import {apiDefault} from '../../api.js';
const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

const initialState = {
  error: ``,
  authorizationStatus: AuthorizationStatus.NO_AUTH,
  user: {
    id: -1,
    name: ``,
    avatar: ``,
    email: ``,
    isPro: false,
  }
};

const ActionType = {
  REQUIRED_AUTHORIZATION: `REQUIRED_AUTHORIZATION`,
  SET_LOADING_ERROR: `SET_LOADING_ERROR`
};

const ActionCreator = {
  requireAuthorization: (status, user) => {
    return {
      type: ActionType.REQUIRED_AUTHORIZATION,
      payload: {status, user}
    };
  },
  setLoadingError: (error) => {
    return {
      type: ActionType.SET_LOADING_ERROR,
      payload: error
    };
  },
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case ActionType.REQUIRED_AUTHORIZATION:
      return extend(state, {
        authorizationStatus: action.payload.status,
        user: action.payload.user
      });
    case ActionType.SET_LOADING_ERROR:
      return extend(state, {
        error: action.payload
      });
  }

  return state;
};

const Operation = {
  checkAuth: () => (dispatch) => {
    return apiDefault.get(`/login`)
      .then((response) => {
        dispatch(ActionCreator.requireAuthorization(AuthorizationStatus.AUTH, adaptUser(response.data)));
      }
      // .catch((err) => {
      //   throw err;
      // });
      , () => dispatch(ActionCreator.requireAuthorization(AuthorizationStatus.NO_AUTH, {id: -1,
        name: ``,
        avatar: ``,
        email: ``,
        isPro: false})));
  },

  login: (authData) => (dispatch, getState, api) => {
    return api.post(`/login`, {
      email: authData.login,
      password: authData.password,
    })
      .then((response) => {
        dispatch(ActionCreator.requireAuthorization(AuthorizationStatus.AUTH, adaptUser(response.data)));
      })
    .catch((err) => {
      const {response} = err;
      dispatch(ActionCreator.setLoadingError(response.data.error));
      // throw err;
    });
  },
};

export {reducer, ActionCreator, ActionType, AuthorizationStatus, Operation};
