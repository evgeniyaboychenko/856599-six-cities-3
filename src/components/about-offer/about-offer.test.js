import React from 'react';
import renderer from 'react-test-renderer';
import AboutOffer from './about-offer.jsx';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';
import NameSpace from '../../reducer/name-space.js';
import {AuthorizationStatus} from '../../reducer/user/user.js';
import {MemoryRouter} from "react-router-dom";

const mockStore = configureStore([]);
jest.mock(`../map/map.jsx`);
jest.mock(`../comment-list/comment-list.jsx`);
jest.mock(`../offer-list/offer-list.jsx`);

const DESCRIPTIONS = [`A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.`, `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`];
const AVATARS = [`avatar-angelina.jpg`, `avatar-max.jpg`];
const NAMES = [`Angelina`, `Max`, `Kate`, `Pol`, `Mike`];
const CITIES = [`Paris`, `Cologne`, `Brussels`, `Amsterdam`, `Hamburg`, `Dusseldorf`];

const offerCard = {
  id: 1,
  cityLocation: {
    coordinates: [48.856663, 2.351556], zoom: 10
  },
  cityName: CITIES[0],
  name: `Beautiful & luxurious studio at great location`,
  rating: 3,
  price: 100,
  type: `apartment`,
  isPremium: false,
  isFavorite: true,
  descriptions: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
  photos: [`img/1.png`, `img/2.png`],
  previewImage: `img/1.png`,
  countRooms: 3,
  maxGuests: 4,
  appliances: [`Heating`, `Kitchen`],
  coordinates: [52.35514938496378, 4.673877537499948],
  zoom: 8,
  owner: {
    avatar: `img/1.png`, id: 3, name: `Angelina`, isSuper: true
  }
};
const offersNear = [
  {
    id: 2,
    cityLocation: {
      coordinates: [48.856663, 2.351556], zoom: 10
    },
    cityName: CITIES[0],
    name: `Beautiful & luxurious studio at great location`,
    rating: 3,
    price: 100,
    type: `apartment`,
    isPremium: false,
    isFavorite: true,
    descriptions: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    photos: [`img/1.png`, `img/2.png`],
    previewImage: `img/1.png`,
    countRooms: 3,
    maxGuests: 4,
    appliances: [`Heating`, `Kitchen`],
    coordinates: [52.35514938496378, 4.673877537499948],
    zoom: 8,
    owner: {
      avatar: `img/1.png`, id: 3, name: `Angelina`, isSuper: true
    }
  },
  {
    id: 3,
    cityLocation: {
      coordinates: [48.856663, 2.351556], zoom: 10
    },
    cityName: CITIES[0],
    name: `Beautiful & luxurious studio at great location`,
    rating: 3,
    price: 100,
    type: `apartment`,
    isPremium: false,
    isFavorite: true,
    descriptions: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    photos: [`img/1.png`, `img/2.png`],
    previewImage: `img/1.png`,
    countRooms: 3,
    maxGuests: 4,
    appliances: [`Heating`, `Kitchen`],
    coordinates: [52.35514938496378, 4.673877537499948],
    zoom: 8,
    owner: {
      avatar: `img/1.png`, id: 3, name: `Angelina`, isSuper: true
    }
  },
  {
    id: 4,
    cityLocation: {
      coordinates: [48.856663, 2.351556], zoom: 10
    },
    cityName: CITIES[0],
    name: `Beautiful & luxurious studio at great location`,
    rating: 3,
    price: 100,
    type: `apartment`,
    isPremium: false,
    isFavorite: true,
    descriptions: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    photos: [`img/1.png`, `img/2.png`],
    previewImage: `img/1.png`,
    countRooms: 3,
    maxGuests: 4,
    appliances: [`Heating`, `Kitchen`],
    coordinates: [52.35514938496378, 4.673877537499948],
    zoom: 8,
    owner: {
      avatar: `img/1.png`, id: 3, name: `Angelina`, isSuper: true
    }
  }
];

const error = ``;
const user = {
  id: 1,
  name: `Nike`,
  avatar: `nike.jpg`,
  email: `nike@n.ru`,
  isPro: false,
};
const authorizationStatus = AuthorizationStatus.AUTH;
const comments =
[{
  id: 1,
  avatar: AVATARS[0],
  name: NAMES[0],
  rating: 2,
  text: DESCRIPTIONS[0],
  date: `2020-03-29T18:07:19.382Z`,
  idUser: 1,
  isPro: false,
},
{
  id: 2,
  avatar: AVATARS[1],
  name: NAMES[1],
  rating: 2,
  text: DESCRIPTIONS[1],
  date: `2020-03-29T18:07:19.382Z`,
  idUser: 1,
  isPro: false,
}];
const isSubmitForm = false;

it(`should AboutOffer render correctly`, () => {
  const store = mockStore({
    [NameSpace.DATA]: {
      offersNear},
    [NameSpace.STATE]: {
      activeSortItem: `Popular`,
    },
    [NameSpace.USER]: {
      user,
      authorizationStatus: AuthorizationStatus.AUTH,
      error: `error`
    },
    [NameSpace.COMMENT]: {
      isSubmitForm: false,
      comments
    }
  });
  const tree = renderer.create(
      <Provider store={store}>
        <MemoryRouter>
          <AboutOffer
            onSubmit = {()=>{}}
            onSubmitDisableButton = {()=>{}}
            isSubmitForm = {isSubmitForm}
            comments = {comments}
            offersNear = {offersNear}
            offerCard = {offerCard}
            user = {user}
            error = {error}
            authorizationStatus = {authorizationStatus}
          />
        </MemoryRouter>
      </Provider>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
