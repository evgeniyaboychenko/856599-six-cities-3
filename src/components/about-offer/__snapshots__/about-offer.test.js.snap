// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should AboutOffer render correctly 1`] = `
<div
  className="page"
>
  <header
    className="header"
  >
    <div
      className="container"
    >
      <div
        className="header__wrapper"
      >
        <div
          className="header__left"
        >
          <a
            className="header__logo-link"
            href="main.html"
          >
            <img
              alt="6 cities logo"
              className="header__logo"
              height="41"
              src="/img/logo.svg"
              width="81"
            />
          </a>
        </div>
        <nav
          className="header__nav"
        >
          <ul
            className="header__nav-list"
          >
            <li
              className="header__nav-item user"
            >
              <a
                className="header__nav-link header__nav-link--profile"
                href="#"
              >
                <div
                  className="header__avatar-wrapper user__avatar-wrapper"
                />
                <span
                  className="header__user-name user__name"
                >
                  Oliver.conner@gmail.com
                </span>
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
  <main
    className="page__main page__main--property"
  >
    <section
      className="property"
    >
      <div
        className="property__gallery-container container"
      >
        <div
          className="property__gallery"
        >
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="/img/room.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="/img/apartment-01.jpg"
            />
          </div>
        </div>
      </div>
      <div
        className="property__container container"
      >
        <div
          className="property__wrapper"
        >
          <div
            className="property__mark"
          >
            <span>
              Premium
            </span>
          </div>
          <div
            className="property__name-wrapper"
          >
            <h1
              className="property__name"
            >
              Canal View Prinsengracht
            </h1>
            <button
              className="property__bookmark-button button"
              type="button"
            >
              <svg
                className="property__bookmark-icon"
                height="33"
                width="31"
              >
                <use
                  xlinkHref="#icon-bookmark"
                />
              </svg>
              <span
                className="visually-hidden"
              >
                To bookmarks
              </span>
            </button>
          </div>
          <div
            className="property__rating rating"
          >
            <div
              className="property__stars rating__stars"
            >
              <span
                style={
                  Object {
                    "width": 40,
                  }
                }
              />
              <span
                className="visually-hidden"
              >
                Rating
              </span>
            </div>
            <span
              className="property__rating-value rating__value"
            >
              4.8
            </span>
          </div>
          <ul
            className="property__features"
          >
            <li
              className="property__feature property__feature--entire"
            >
              apartment
            </li>
            <li
              className="property__feature property__feature--bedrooms"
            >
              2
               Bedrooms
            </li>
            <li
              className="property__feature property__feature--adults"
            >
              Max 
              2
               adults
            </li>
          </ul>
          <div
            className="property__price"
          >
            <b
              className="property__price-value"
            >
              €
              100
            </b>
            <span
              className="property__price-text"
            >
               night
            </span>
          </div>
          <div
            className="property__inside"
          >
            <h2
              className="property__inside-title"
            >
              What's inside
            </h2>
            <ul
              className="property__inside-list"
            >
              <li
                className="property__inside-item"
              >
                Wi-Fi
              </li>
              <li
                className="property__inside-item"
              >
                Washing machine
              </li>
              <li
                className="property__inside-item"
              >
                Towels
              </li>
              <li
                className="property__inside-item"
              >
                Heating
              </li>
            </ul>
          </div>
          <div
            className="property__host"
          >
            <h2
              className="property__host-title"
            >
              Meet the host
            </h2>
            <div
              className="property__host-user user"
            >
              <div
                className="property__avatar-wrapper property__avatar-wrapper--pro user__avatar-wrapper"
              >
                <img
                  alt="Host avatar"
                  className="property__avatar user__avatar"
                  height="74"
                  src="/img/avatar-angelina.jpg"
                  width="74"
                />
              </div>
              <span
                className="property__user-name"
              >
                Angelina
              </span>
            </div>
            <div
              className="property__description"
            >
              <p
                className="property__text"
              >
                A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam
              </p>
            </div>
          </div>
        </div>
      </div>
      <section
        className="property__map map"
      />
    </section>
    <div
      className="container"
    >
      <section
        className="near-places places"
      >
        <h2
          className="near-places__title"
        >
          Other places in the neighbourhood
        </h2>
        <div
          className="near-places__list places__list"
        />
      </section>
    </div>
  </main>
</div>
`;
